project/CrossJava.scala

```diff
@@ -30,10 +29,20 @@ case class JavaVersion(numbers: Vector[Long], vendor: Option[String]) {
   }
 }
 object JavaVersion {
+
+  val specificationVersion: String = sys.props("java.specification.version")
+  val version: String = sys.props("java.version")
+  val is1x: Boolean = version.startsWith("1.")
+
   def apply(version: String): JavaVersion = CrossJava.parseJavaVersion(version)
   def apply(numbers: Vector[Long], vendor: String): JavaVersion = new JavaVersion(numbers, Option(vendor))
-}
 
+  def notOnJdk8[T](values: Seq[T]): Seq[T] = if (is1x) Seq.empty[T] else values
```

Comment:
I'd prefer the if-else at the call site, with this I'd have to go here to see what it does if not etc.

Reply:
Actually I like having these helpers: moving the if-else to the call site often introduces the need for additional parentheses and indentation. This makes it a lot easier to read in those cases.

Reply:
I know mean @johanandren, though if It's a split vote, I'd rather keep it this new way so we gain the benefits of 
* No code duplication (we duplicated 4 lines, twice) 
* Improved readability (it's much cleaner)
* Having the java version related logic encapsulated in the pre-existing `JavaVersion` object vs sprinkled around
* Faster to maintain when / if changes to that one piece of logic need updating, we update in one place vs several files

Reply:
Sure, keep it.